metals/src/main/scala/scala/meta/internal/bsp/BspConnector.scala

```diff
@@ -165,18 +151,37 @@ class BspConnector(
       createBloopAndConnect: () => Future[BuildChange]
   ): Future[Boolean] = {
     val bloopPresent = buildTools.isBloop
+    val bloopIsAnOption =
+      bloopPresent || buildTools.loadSupported().nonEmpty
     bspServers.findAvailableServers() match {
+      case Nil if bloopIsAnOption =>
+        client.showMessage(BspSwitch.onlyOneServer(name = "Bloop"))
+        Future.successful(false)
       case Nil =>
-        if (bloopPresent)
-          client.showMessage(BspSwitch.onlyOneServer(name = "Bloop"))
-        else
-          client.showMessage(BspSwitch.noInstalledServer)
+        client.showMessage(BspSwitch.noInstalledServer)
+        Future.successful(false)
+      case singleServer :: Nil if !bloopIsAnOption =>
+        client.showMessage(
+          BspSwitch.onlyOneServer(name = singleServer.getName())
+        )
         Future.successful(false)
-      case availableServers =>
-        val currentBsp = tables.buildServers.selectedServer()
-        askUser(availableServers, currentBsp).map {
+      case multipleServers =>
+        val currentSelectedServer = tables.buildServers.selectedServer()
+        val allPossibleConnections =
+          if (bloopIsAnOption)
+            new BspConnectionDetails(
+              "Bloop",
+              ImmutableList.of(),
+              userConfig().currentBloopVersion,
+              "",
+              ImmutableList.of()
+            ) :: multipleServers
+          else multipleServers
+        askUser(allPossibleConnections, currentSelectedServer).map {
           case ResolvedBloop
-              if currentBsp.contains(BspConnector.BLOOP_SELECTED) =>
+              if currentSelectedServer.contains(
+                BspConnector.BLOOP_SELECTED
+              ) =>
```

Comment:
This pattern-match might be a little bit simplified by adding bloop option into the available server list before matching.
Smth like:
```scala
val availableServers = {
  val found = bspServers.findAvailableServers
  if (bloopIsAnOption) {
     new BspConnectionDetails(
          "Bloop",
           ImmutableList.of(),
           userConfig().currentBloopVersion,
           "",
           ImmutableList.of()
       ) :: found
  } else found
}
availableServers match {
   case Nil =>
        client.showMessage(BspSwitch.noInstalledServer)
        Future.successful(false)
    case singleServer :: Nil  =>
        client.showMessage(
          BspSwitch.onlyOneServer(name = singleServer.getName())
        )
        Future.successful(false)
     case multipleServers =>
        val currentSelectedServer = tables.buildServers.selectedServer()
        askUser(multipleServers, currentSelectedServer)
         ...
}
```

Reply:
ah, great suggestion! Fixed and simplified.